src/community/s_metric.jl

```diff
@@ -0,0 +1,35 @@
+"""
+    s_metric(g; norm=true)
+
+Return the normalised s-metric of `g`.
+
+The s-metric is defined as the sum of the product of degrees between pair of vertices
+for every edge in `g`. [Ref](https://arxiv.org/abs/cond-mat/0501169)
+In directed graphs, the paired values are the out-degree of source vertices
+and the in-degree of destination vertices.
+It is normalised by the maximum s_metric obtained from the family of graph
+with similar degree distribution. s_max is computed from an approximation
+formula as in https://journals.aps.org/pre/pdf/10.1103/PhysRevE.75.046102
+If `norm=false`, no normalisation is performed.
+
+# Examples
+```jldoctest
+julia> using LightGraphs
+
+julia> s_metric(star_graph(4))
+0.6
+```
+"""
+
+function s_metric(g::AbstractGraph{T}; norm=true) where T
+    s = zero(T)
```

Comment:
Maybe `zero(T)` is not the right type here. For small graphs it might be something like `Int8` that would overflow quite soon. Maybe it is best, to use `Int` here, or `Float64`, then the resulting type would be the same no matter if `norm` is set to `true` or `false`.